packages/aws-cdk-lib/aws-batch/lib/job-queue.ts

```diff
@@ -135,6 +143,85 @@ export interface OrderedComputeEnvironment {
   readonly order: number;
 }
 
+/**
+ * Specifies an action that AWS Batch will take after the job has remained at
+ * the head of the queue in the specified state for longer than the specified time.
+ */
+export interface JobStateTimeLimitAction {
+  /**
+   * The action to take when a job is at the head of the job queue in the specified state
+   * for the specified period of time.
+   *
+   * @default JobStateTimeLimitActionsAction.CANCEL
+   */
+  readonly action?: JobStateTimeLimitActionsAction;
+
+  /**
+   * The approximate amount of time, that must pass with the job in the specified
+   * state before the action is taken.
+   *
+   * The minimum value is 10 minutes and the maximum value is 24 hours.
+   */
+  readonly maxTime: Duration;
+
+  /**
+   * The reason to log for the action being taken.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/userguide/troubleshooting.html#job_stuck_in_runnable
+   */
+  readonly reason: JobStateTimeLimitActionsReason;
```

Comment:
I'm not too familiar with batch queue, but reading the cfn documentation, it seems that the reason can be arbitrary string and it doesn't mention that it has to be due to time limit. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-jobqueue-jobstatetimelimitaction.html#cfn-batch-jobqueue-jobstatetimelimitaction-action

Reply:
@GavinZZ 
Is the intention that the name `JobStateTimeLimitActionsReason` is inappropriate?
Here, I named it this way because it is the 'Reason' within the 'JobStateTimeLimitActions' property.
This is because Reason is already defined separately within aws-batch.
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_batch.Reason.html